src/msw/textctrl.cpp

```diff
@@ -2861,7 +2866,63 @@ bool wxTextCtrl::MSWSetCharFormat(const wxTextAttr& style, long start, long end)
             cf.dwEffects |= CFE_STRIKEOUT;
         }
     }
-
+    if( style.HasFontUnderline() )
+    {
+        cf.dwMask |= CFM_UNDERLINETYPE;
+        wxTextAttrUnderlineType underlineType = style.GetUnderlineType();
+        switch( underlineType )
+        {
+            case wxTEXT_ATTR_UNDERLINE_SOLID:
+                cf.bUnderlineType = CFU_UNDERLINE;
+                break;
+            case wxTEXT_ATTR_UNDERLINE_DOUBLE:
+                cf.bUnderlineType = CFU_UNDERLINEDOUBLE;
+                break;
+            case wxTEXT_ATTR_UNDERLINE_WAVE:
+                cf.bUnderlineType = CFU_UNDERLINEWAVE;
+                break;
+        }
+#if _RICHEDIT_VER >= 0x0800
+// The colours are coming from https://docs.microsoft.com/en-us/windows/desktop/api/tom/nf-tom-itextdocument2-geteffectcolor.
+// Not all values from wxTheColourDatabase are supported as can be seen from the code
```

Comment:
Obsolete comment.

Reply:
@MaartenBent ,
I will.
This is a reminder to mention this in the docs.